### ISSUE TITLE
Google Keyboard searching issue

### ISSUE THREAD

**Issue by kunalverma25** [brave/browser-ios#1080]
- Did you search for similar issues before submitting this one?
Yes

- Description:
Google Keyboard don't recognise autocomplete URL, instead searched for entered text.

- Device (iPhone5, iPhone6s plus, iPad 3, ?):
All

- Brave Version:
1.4.1

- Steps to reproduce:
    1. Install Google Keyboard
    2. Enter text like "goo", which results in prediction like "google.com"
    3. Press Go/Enter

- Actual Behavior
Search on google for "goo"

- Expected Behavior
Go to google.com

- Is this an issue with Beta build?
Yes

- Is this an issue in the currently released version?
Yes

- Can this issue be consistently reproduced?
Yes

- Extra QA steps:
    1. 
    2. 
    3.

- Website problems only: 
- did you check with Brave Shields down? 
- did you check in Puffin/Dolphin/iCab/Firefox (UIWebView-based browsers)? 

- Screenshot if needed:

- Any related issues:

**Comment by srirambv** [comment]
@kunalverma25  Irrespective of any keyboard when  I type goo and tap on Go, with the suggestion google.com it opens google.com home page. Doesn't search for goo. This is both on 1.4.1 and 1.4.2. 

Is there any step that I am missing?

**Comment by kunalverma25** [comment]
@srirambv
The issue only appears in google keyboard.
You have to use the Gboard app and switch to that.
The issue was in 1.4 release.
I think after that it was merged with master.

**Comment by LaurenWags** [comment]
@kunalverma25 I believe the fix worked, this is what I see. Can you confirm this is your expected result?
![1080ios](https://user-images.githubusercontent.com/28145373/30219721-f7c99de8-948a-11e7-8d1f-eba02d83a9e1.gif)

**Comment by kjozwiak** [comment]
Installed the Google Keyboard and reproduced the issue using `1.4.1 (17.08.25.08)` using the original STR mentioned above. Once I typed in "goo" and selected "Go/Enter", Brave searched for the term "goo" rather than loading google.com. I also tried a few other terms like "duckduckgo", "bing", "Yaho" etc.. Rather than loading the appropriate websites, the terms were being searched using Google.

Using `1.4.2 (17.09.07.20)`, I couldn't reproduce the above issues. Brave correctly loaded the websites rather than searching for the terms.

**Device Used:**

* iPhone 6s+ using iOS 10.3.3 - PASSED

**Build Used:**

* `1.4.1 (17.08.25.08)` - Reproduced the issue
* `1.4.2 (17.09.07.20)` - Couldn't reproduce the issue (FIXED)
